pkg/kubectl/cmd/run.go

```diff
@@ -196,7 +196,12 @@ func Run(f *cmdutil.Factory, cmdIn io.Reader, cmdOut, cmdErr io.Writer, cmd *cob
 	}
 
 	params["env"] = cmdutil.GetFlagStringSlice(cmd, "env")
-
+	port := cmdutil.GetFlagInt(cmd, "port")
```

Comment:
Validation for the deployment should complain on --port=0. Can you post the yaml of the deployment that is created when you run the second cmd?